frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -681,17 +670,17 @@ function ReaderBookmark:toggleBookmark(pn_or_xp)
         table.remove(self.bookmarks, index)
     else
         -- build notes from TOC
-        local notes = self.ui.toc:getTocTitleByPage(pn_or_xp)
-        local chapter_name = notes
-        if notes ~= "" then
+        local notes = "★" -- distinguish bookmarks from highlights and notes
+        local chapter_name = self.ui.toc:getTocTitleByPage(pn_or_xp)
+        if chapter_name ~= "" then
             -- @translators In which chapter title (%1) a note is found.
-            notes = T(_("in %1"), notes)
+            notes = notes .. " " .. T(_("in %1"), chapter_name)
```

Comment:
This I don't like - and you broke your contract of not updating the datastructure :/

This star is cosmectic/display stuff - you should not have to update it in the datastructure - because then, it becomes just what "Page N  @ date" is in highlights: crap we will have to parse and clean.
You should keep saving page bookmarks how they are, and find a way to distinguish them at display time, and add the stars into what's being used for display purpose (your `item_table` clone).

(And may be `"notes="` has no purpose for page-bookmarks and may be it won't even be used for migrating - but I don't know, and don't want to be forced to investigate that now :)